### ISSUE TITLE
Set tail logs value to -1 as a workaround for VK truncated logs

### ISSUE THREAD

**Issue by radu-matei** [brigadecore/brigade#952]
Because of https://github.com/virtual-kubelet/virtual-kubelet/issues/541 (actual context here), the logs for any job running on top of VK are truncated to 10. We could set the tail to -1 in all places where we request logs from Kubernetes.

This change would allow usable logs for VK jobs, without depending on the upstream change, and would have minimal impact otherwise (as by default, the kubelet implementation returns all logs).

Thoughts?

**Comment by mumoshu** [comment]
I have no specific rationale for keeping it dependent on a kubelet implementation. Our users should be able to retrieve full build logs on any K8s cluster with any kubelet implementation, as long as the effort to make it kubelet-impl-agnostic isn't so much.

With that said - sounds good! Let's set `-1` everywhere.

**Comment by mumoshu** [comment]
@radu-matei I tried literally setting `-1` but that ended up with a validation error in client-go:

```
$ go run ./brig/cmd/brig run -e ping myuser/myrepo
Event created. Waiting for worker pod named "brigade-worker-01dhn57p6emgwkfs1dkpskatwa".
Build: 01dhn57p6emgwkfs1dkpskatwa, Worker: brigade-worker-01dhn57p6emgwkfs1dkpskatwa
Error: PodLogOptions "brigade-worker-01dhn57p6emgwkfs1dkpskatwa" is invalid: tailLines: Invalid value: -1: must be greater than or equal to 0
exit status 1
```

https://github.com/mumoshu/brigade/commit/16baa025ce241bdda7ed139a5cd23296fd98814d

The API type says that it is int64. So in theory setting [`math.MaxInt64`](https://golang.org/src/math/const.go) would work, assuming we are unlikely to read more than `9223372036854775807 ` lines of log. (Does k8s even support it? 😃 

---

But after asking myself, I'm wondering if we can add options for various brigade components to allow specifying MAXIMUM number of log lines to tail, and a sensible default value that is lower than `9223372036854775807`.

Proposed flags and options:

1. `--logs-tail-lines M` for `brigade-api`, so that job logs in kashti isn't truncated
2. `--tail-lines M` a.k.a `-n M` for `brig`, so that `brig build logs` output isn't truncated
3. `Job.LogsTailLines = N` for [Job](https://github.com/brigadecore/brigade/blob/6f2bf49a37bb92770459d9e80400b4e67338e521/brigade-worker/src/k8s.ts#L478-L487) class, so that jog logs emitted and captured while running `brigade.js`/streamed to the worker aren't truncated. (I considered adding `logs({tailLines: N)` first but I think that's an option/just a nice-to-have in respect to the goal of this issue

Proposed sensible defaults:

- For 3. use N =  `100000` ("believing" you won't usually read 100000 lines of logs per job 😃 
- 1. 2. should have the same default value `1000000000`, that is M times of N of 3., where M is the number of jobs no more than we "believe" to be run within a single Brigade build, say, 10000

---

Maybe we can just proceed with just hard-coding `math.MaxInt64`, and then discuss about the options and sensible defaults in an another GitHub issue.

WDYT?

**Comment by radu-matei** [comment]
I think this was fixed by #972
